serverless/serverless
# Added API Keys to info plugin

## What did you implement:

***Implementing Issue:*** #2058

This PR adds the API Keys you have in your config along with their actual values to the info plugin. So when you deploy or run `sls info` you'll be able to know the values of the API Keys.

## How did you implement it:
I added a new output template for each API key template we have. Then the info plugin fetches those outputs and display the values.

## How can we verify it:

Run `serverless deploy` with the following config:

```yml
service: serverless-api-keys

provider:
  name: aws
  runtime: nodejs4.3
  apiKeys:
    - myAPI
    - yourAPI

functions:
  hello:
    handler: handler.hello
    events:
      - http: get users/create
```

With **master** you should see this:

```
Serverless: Stack update finished...

Service Information
service: serverless-api-keys
stage: dev
region: us-east-1
endpoints:
  GET - https://xxx.execute-api.us-east-1.amazonaws.com/dev/users/create
functions:
  serverless-api-keys-dev-hello: arn:aws:lambda:us-east-1:xxx:function:serverless-api-keys-dev-hello
```
With **PR** you should see this:

```
Serverless: Stack update finished...

Service Information
service: serverless-api-keys
stage: dev
region: us-east-1
api keys:
  yourAPI: 8FiGK0hrhcDmM1IOUh8i8iQUG5ZgsES7F1OeIBs6
  myAPI: HYQZXgCXNk15NrfHYkl3W3927Zs9f89x5VeSGzCJ
endpoints:
  GET - https://xxx.execute-api.us-east-1.amazonaws.com/dev/users/create
functions:
  serverless-api-keys-dev-hello: arn:aws:lambda:us-east-1:xxx:function:serverless-api-keys-dev-hello
```
## Todos:
